service/controller/resource/statuscondition/create.go

```diff
@@ -106,11 +106,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 }
 
 func (r *Resource) computeCreateClusterStatusConditions(ctx context.Context, cr infrastructurev1alpha2.CommonClusterObject, nodes []corev1.Node, machineDeployments []apiv1alpha2.MachineDeployment) error {
-	cc, err := controllercontext.FromContext(ctx)
+	componentVersions, err := r.releaseVersion.ComponentVersion(ctx, cr)
```

Comment:
I was assuming that using the infrastructurev1alpha2.CommonClusterObject can also be used to get the belonging release but maybe this is wrong. This needs to be clarified

Reply:
I think it is ok since the common cluster thingy is an interface implementing the labels getters, which is what we need. See https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/infrastructure/v1alpha2/common_cluster_status_spec.go#L11. We could maybe think of a better interface that our service implementations require to be injected when fetching information so that we have more type safety. Right now we just say `interface{}` which removes the compiler support.